pandas/tests/groupby/aggregate/test_other.py

```diff
@@ -666,3 +666,28 @@ def weird_func(x):
 
     result = df["decimals"].groupby(df["id1"]).agg(weird_func)
     tm.assert_series_equal(result, expected, check_names=False)
+
+
+def test_groupby_agg_boolean_dype():
+    # GH Issue #58031
+    # Ensure return type of aggregate dtype has consistent behavior
+    # for 'bool' and 'boolean' because boolean not covered under numpy
+
+    df_boolean = DataFrame({"0": [1, 2, 2], "1": [True, True, None]})
+    df_boolean[1] = df_boolean["1"].astype("boolean")
+
+    df_bool = DataFrame({"0": [1, 2, 2], "1": [True, True, None]})
+    df_bool[1] = df_bool["1"].astype("bool")
+
+    boolean_return_type = (
+        df_boolean.groupby("0")
+        .aggregate(lambda s: s.fillna(False).mean())
+        .dtypes.values[0]
+    )
+    bool_return_type = (
+        df_bool.groupby("0")
+        .aggregate(lambda s: s.fillna(False).mean())
+        .dtypes.values[0]
+    )
+
+    assert boolean_return_type == bool_return_type
```

Comment:
When writing tests, always test the full result (using `tm.assert_frame_equals` here) instead of just one part of it. See the tests above for examples.